frontend/views/containers/proposals/AddMembers.vue

```diff
@@ -111,67 +99,51 @@ export default {
-          this.L('{searchUser} is already on the group', { searchUser })
-        )
-      }
-
-      try {
-        const contractID = await sbp('namespace/lookup', searchUser)
-        const userState = await sbp('state/latestContractState', contractID)
-        this.ephemeral.invitesToSend[index] = { contractID, ...userState }
-        Vue.set(this.form.eachFeedbackMsg, index, {
-          text: this.L('Ready to be invited!'),
-          class: 'has-text-success'
-        })
+    inviteeUpdate (e, index) {
+      if (e.target.value.length > 0 && !this.ephemeral.isValid) {
         this.ephemeral.isValid = true
-      } catch (err) {
-        console.log(err)
-        this.setInviteError(index, this.L('Non existing user. TODO support general invites'))
       }
-    },
-    setInviteError (index, text) {
-      this.ephemeral.invitesToSend[index] = null
-      Vue.set(this.form.eachFeedbackMsg, index, {
-        text,
-        class: 'error'
-      })
+      if (e.target.value.length === 0 && this.ephemeral.invitesCount === 1 && this.ephemeral.isValid) {
+        this.ephemeral.isValid = false
+      }
     },
     removeInvitee (index) {
       this.ephemeral.invitesCount -= 1
       this.form.invitees.splice(index, 1)
-      this.ephemeral.invitesToSend[index] = null
     },
-    addInviteeSlot () {
+    addInviteeSlot (e) {
       this.ephemeral.invitesCount += 1
+      Vue.nextTick(() => {
+        const inviteeSlots = this.$refs.fieldset.getElementsByTagName('label')
+        const newInviteeSlot = inviteeSlots[inviteeSlots.length - 1]
+        newInviteeSlot && newInviteeSlot.focus()
+      })
     },
     async submit (form) {
-      try {
+      const invitationsSent = []
+      let hasFailed = false
+      // NOTE: All invitees proposals will expire at the exact same time.
+      // That plus the proposal creator is what we'll use to know
+      // which proposals should be displayed visually together.
+      const expiresDateMs = Date.now() + this.groupSettings.proposals[PROPOSAL_INVITE_MEMBER].expires_ms
+
+      for (const inviteeName of this.form.invitees) {
         const groupId = this.currentGroupId
-        const groupName = this.groupSettings.groupName
-        // NOTE: All invitees proposals will expire at the exact same time.
-        // That plus the proposal creator is what we'll use to know
-        // which proposals should be displayed visually together.
-        const expiresDateMs = Date.now() + this.groupSettings.proposals[PROPOSAL_INVITE_MEMBER].expires_ms
-        for (const index in this.ephemeral.invitesToSend) {
-          const member = this.ephemeral.invitesToSend[index]
-          const mailbox = member.attributes.mailbox
-          const memberName = member.attributes.name
+        let identityContract
+        try {
+          const contractID = await sbp('namespace/lookup', inviteeName)
+          identityContract = await sbp('state/latestContractState', contractID)
+        } catch (e) {
```

Comment:
TIL: When using async/await inside a loop, use _for of / for in_. Array.forEach doesn't support async await correctly out of the box. cc @pieer @SebinSong